pjsip/src/pjsua-lib/pjsua_call.c

```diff
@@ -4147,6 +4171,452 @@ static pj_status_t process_pending_reinvite(pjsua_call *call)
 }
 
 
+static void trickle_ice_retrans_18x(pj_timer_heap_t *th,
+				    struct pj_timer_entry *te)
+{
+    pjsua_call *call = (pjsua_call*)te->user_data;
+    pjsip_tx_data *tdata;
+    pj_time_val delay;
+
+    PJ_UNUSED_ARG(th);
+
+    /* If trickling has been started or dialog has been established on
+     * both sides, stop 18x retransmission.
+     */
+    if (call->trickle_ice.trickling || call->trickle_ice.remote_dlg_est)
+	return;
+
+    /* Make sure last tdata is 18x response */
+    tdata = call->inv->invite_tsx->last_tx;
+    if (tdata->msg->type != PJSIP_RESPONSE_MSG ||
+	tdata->msg->line.status.code/10 != 18)
+    {
+	return;
+    }
+
+    /* Retransmit 18x */
+    ++call->trickle_ice.retrans18x_count;
+    PJ_LOG(4,(THIS_FILE,
+	      "Call %d: ICE trickle retransmitting 18x (retrans #%d)",
+	      call->index, call->trickle_ice.retrans18x_count));
+
+    pjsip_tx_data_add_ref(tdata);
+    pjsip_tsx_retransmit_no_state(call->inv->invite_tsx, tdata);
+
+    /* Schedule next retransmission */
+    if (call->trickle_ice.retrans18x_count < 6) {
+	delay.sec = 0;
+	delay.msec = (1 << call->trickle_ice.retrans18x_count)*
+		     pjsip_cfg()->tsx.t1;
```

Comment:
From Code scanning
"Multiplication result converted to larger type
Multiplication result may overflow 'unsigned int' before it is converted to 'long'.
"

Reply:
not sure the correct solution, will just copy the pattern of [this code](https://github.com/pjsip/pjproject/blob/master/pjsip/src/pjsip/sip_transaction.c#L2345).